density.c

```diff
@@ -418,8 +418,11 @@ density_postprocess(int i, TreeWalk * tw)
                 SPHP(i).DhsmlEgyDensityFactor *= -SPHP(i).DhsmlDensityFactor;
                 SPHP(i).EgyWtDensity /= EntPred;
             } else {
-                SPHP(i).DhsmlEgyDensityFactor=0;
-                SPHP(i).EgyWtDensity=0;
+                message(1, "Something weird: Particle %d (ID %ld): Entropy %g, EgyWtDensity %g\n",
```

Comment:
Not that weird -- the first initialization density EntPred and EgyWtDensity are always zero. This protects against that -- do you see a horde of warning messages after this when hydro starts?

Reply:
Ah indeed! Well noticed. I will amend the commit.

Reply:
When I am ready to merge - for now it is proving a useful warning.